### PR TITLE
Add formatter for lua 5.1

### PR DESCRIPTION
**Pull request by ElPiloto**
luaformatterfiveone is a formatter for Lua 5.1.  It is readily available
through standard via `luarocks install formatterfiveone`.

### CODE REVIEW

**Comment by ElPiloto** [comment]
@googlebot I fixed it

**Comment by ElPiloto** [comment]
@googlebot I fixed it

**Comment by ElPiloto** [comment]
@googlebot I fixed it

File: plugin/register.vim [hunk 1]
```diff
@@ -39,3 +39,4 @@ call s:registry.AddExtension(codefmt#shfmt#GetFormatter())
 call s:registry.AddExtension(codefmt#zprint#GetFormatter())
 call s:registry.AddExtension(codefmt#fish_indent#GetFormatter())
 call s:registry.AddExtension(codefmt#nixpkgs_fmt#GetFormatter())
+call s:registry.AddExtension(codefmt#luaformatterfiveone#GetFormatter())
```
Comment (dbarnett) [inline plugin/register.vim:42]: Can you add it to the list of formatters at https://github.com/google/vim-codefmt/blob/6761a211eb8690e84167a1cb2efb11f31e07973d/autoload/codefmt.vim#L28 and see if you can get the helpfile regenerated so it shows up as an available formatter in the docs?

File: autoload/codefmt/luaformatterfiveone.vim [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        let l:fname_pattern = 'stdin'
+        let l:tokens = matchlist(line,
+            \ '\C\v^\[string "isCodeValid"\]:(\d+): (.*)')
+        if !empty(l:tokens)
+          call add(l:errors, {
+              \ "filename": @%,
+              \ "lnum": l:tokens[1],
+              \ "text": l:tokens[2]})
+        endif
+              "\ "col": l:tokens[2],
```
Comment (dbarnett) [inline autoload/codefmt/luaformatterfiveone.vim:61]: Stray line here. Is "col" needed? If not, delete.

File: autoload/codefmt/luaformatterfiveone.vim [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    return executable(s:plugin.Flag('luaformatterfiveone_executable'))
+  endfunction
+
+  function l:formatter.AppliesToBuffer() abort
+    return &filetype is# 'lua'
+  endfunction
+
+  ""
+  " Reformat the current buffer with luaformatterfiveone or the binary named in
+  " @flag(luaformatterfiveone)
```
Comment (dbarnett) [inline autoload/codefmt/luaformatterfiveone.vim:39]: This was supposed to be `@flag(luaformatterfiveone_executable)`, right?

**Review by dbarnett — CHANGES_REQUESTED** [review]
Thanks, and sorry for the hassle of whatever was happening w/ the CLA bot there.

File: autoload/codefmt/luaformatterfiveone.vim (hunk 3 above)
Comment (ElPiloto) [reply autoload/codefmt/luaformatterfiveone.vim]: Done.

File: autoload/codefmt/luaformatterfiveone.vim (hunk 2 above)
Comment (ElPiloto) [reply autoload/codefmt/luaformatterfiveone.vim]: Done

File: plugin/register.vim (hunk 1 above)
Comment (ElPiloto) [reply plugin/register.vim:42]: Done

**Review by dbarnett — APPROVED** [review]
Thanks! Merging now.

**Comment by mxngls** [comment]
I followed the installation steps but when trying to format Lua code I get the following error message:
```
Formatter "luaformatterfiveone" is not available. Setup instructions: Install luaformatterfiveone with luarocks. (https://luarocks.org/modules/ElPiloto/formatterfiveo
ne).
```

The package as itself is installed as described in the [linked](https://github.com/ElPiloto/luaformatterfiveone) repo:
```
max in ~  -> luarocks list

Rocks installed for Lua 5.1
---------------------------

formatterfiveone
   0.1.2-1 (installed) - /Users/max/.luarocks/lib/luarocks/rocks-5.1

luafilesystem
   1.6.3-2 (installed) - /Users/max/.luarocks/lib/luarocks/rocks-5.1

metalua-compiler
   0.7.3-1 (installed) - /Users/max/.luarocks/lib/luarocks/rocks-5.1

metalua-parser
   0.7.3-2 (installed) - /Users/max/.luarocks/lib/luarocks/rocks-5.1

penlight
   0.9.8-1 (installed) - /Users/max/.luarocks/lib/luarocks/rocks-5.1
```

Help would be much appreciated.

Thanks in advance.

**Comment by dbarnett** [comment]
Yeah, so [the availability check](https://github.com/google/vim-codefmt/blob/64ffe0761b9499f15ea8b56b153644c488b5bf74/autoload/codefmt/luaformatterfiveone.vim#L30) is just checking if flag path is executable. You don't actually have to change the flag to specify the full path, the default value should work fine for most people, but if you haven't installed it to your path so that you can run `luaformatterfiveone` in the same terminal you started vim from, it won't show as available.

**Comment by nakulj** [comment]
@mxngls thanks! For anyone else running into problems, two tips:

1. If you have a different version of lua installed, tell luarocks to use 5.1 like so: ```bash
luarocks install --lua-dir=/opt/homebrew/opt/lua@5.1/  formatterfiveone
```
2. You want to <pre>luarocks install check<b>s</b></pre> - there's a missing 's' in the original comment.

**Pull request merged**